### PR TITLE
remove jquery dependency

### PR DESCRIPTION
**Pull request by cormac**
for https://github.com/intercom/ember-href-to/issues/2
`matches` isn't supported by phantomJS :disappointed:

### CODE REVIEW

File: addon/helpers/href-to.js [hunk 1]
```diff
@@ -12,28 +12,44 @@ function _getNormalisedRootUrl(router) {
...
-    var handleClick = (e.which === 1 && !e.ctrlKey && !e.metaKey);
 
-    if(handleClick && !$target.hasClass('ember-view') && Em.isNone($target.attr('data-ember-action'))) {
-      var router = getRouter();
+  document.getElementsByTagName('body')[0].onclick = function(e) {
+    var target;
+    if (e.path) {
+      target = _getLinkElementFromPath(e.path);
+    } else {
+      target = e.target;
```
Comment (cormac) [inline addon/helpers/href-to.js]: tests pass a jQuery element

File: addon/helpers/href-to.js [hunk 2]
```diff
@@ -12,28 +12,44 @@ function _getNormalisedRootUrl(router) {
...
 function _setupClickHandler() {
-  Em.$(document.body).on('click', 'a', function(e) {
-    var $target = Em.$(e.currentTarget);
-    var handleClick = (e.which === 1 && !e.ctrlKey && !e.metaKey);
 
-    if(handleClick && !$target.hasClass('ember-view') && Em.isNone($target.attr('data-ember-action'))) {
-      var router = getRouter();
+  document.getElementsByTagName('body')[0].onclick = function(e) {
+    var target;
+    if (e.path) {
```
Comment (cormac) [inline addon/helpers/href-to.js]: no work in FF

File: addon/helpers/href-to.js [hunk 3]
```diff
@@ -12,28 +12,43 @@ function _getNormalisedRootUrl(router) {
...
-  Em.$(document.body).on('click', 'a', function(e) {
-    var $target = Em.$(e.currentTarget);
-    var handleClick = (e.which === 1 && !e.ctrlKey && !e.metaKey);
 
-    if(handleClick && !$target.hasClass('ember-view') && Em.isNone($target.attr('data-ember-action'))) {
-      var router = getRouter();
+  document.getElementsByTagName('body')[0].onclick = function(e) {
+    var target = _parentLinkNode(e.target);
+    if (target) {
+    var handleClick = (e.which === 1 && !e.ctrlKey && !e.metaKey);
```
Comment (paddyobrien) [inline addon/helpers/href-to.js]: `  `

File: addon/helpers/href-to.js [hunk 4]
```diff
@@ -12,28 +12,43 @@ function _getNormalisedRootUrl(router) {
...
+      if(handleClick && !_classContainsEmberView(target.className) && Em.isNone(target.getAttribute('data-ember-action'))) {
```
Comment (paddyobrien) [inline addon/helpers/href-to.js]: perhaps `_isEmberView(target)` would be neater.

**Comment by GavinJoyce** [comment]
@cormac what's the status of this?

**Comment by cormac** [comment]
it was forgotten about :disappointed:, never got a :ship:

**Comment by GavinJoyce** [comment]
ah, the title still has wip in it. I'll review this soon if it's good to go?

**Comment by cormac** [comment]
good to go :smile:

**Comment by GavinJoyce** [comment]
image

**Pull request closed without merging**